test/jit/test_peephole.py

```diff
@@ -529,3 +529,28 @@ def foo(x: int):
         FileCheck().check("graph").check_next("return").run(foo.graph)
         self.assertEqual(foo(2), 2)
         self.assertEqual(foo(1), 1)
+
+    def test_peephole_len_list(self):
+        @torch.jit.script
+        def foo(x):
+            return len(x.size())
+
+        self.run_pass("peephole", foo.graph)
+        FileCheck().check("aten::len").run(foo.graph)
+        inputs = list(foo.graph.inputs())
+        inputs[0].setType(inputs[0].type().with_sizes([None, None]))
+        self.run_pass("peephole", foo.graph)
```

Comment:
should `peephole` be running shape ops by default?

Reply:
yes, it is only the weird pipeline in profiling_executor where we shouldnt